openvdb/points/PointMove.h

```diff
@@ -509,13 +535,41 @@ struct GlobalMovePointsOp
         IndexArray& offsets, const IndexTripleArray& indices) const
     {
         auto& targetArray = mTargetHandles.getArray(targetOffset);
+        targetArray.loadData();
+        targetArray.expand();
+
+        // build a sorted index vector that references the indices in order of their source
+        // leafs and voxels to ensure determinism in the resulting point orders
+
+        std::vector<int> sortedIndices(indices.size());
+        std::iota(std::begin(sortedIndices), std::end(sortedIndices), 0);
+        std::sort(std::begin(sortedIndices), std::end(sortedIndices),
+            [&](int i, int j)
+            {
+                const Index& indexI0(std::get<0>(indices[i]));
+                const Index& indexJ0(std::get<0>(indices[j]));
+                if (indexI0 < indexJ0)          return true;
+                if (indexI0 > indexJ0)          return false;
+                return std::get<2>(indices[i]) < std::get<2>(indices[j]);
+            }
+        );
+
+        for (const auto& index : sortedIndices) {
+            const auto& it = indices[index];
 
-        for (const auto& it : indices) {
             const auto& sourceArray = mSourceHandles.getConstArray(std::get<0>(it));
-            const Index tgtOffset = indexOffsetFromVoxel(std::get<1>(it), targetLeaf, offsets);
-            OPENVDB_NO_DEPRECATION_WARNING_BEGIN
-            targetArray.set(tgtOffset, sourceArray, std::get<2>(it));
-            OPENVDB_NO_DEPRECATION_WARNING_END
+
+            const Index sourceOffset = std::get<2>(it);
+            const Index targetOffset = indexOffsetFromVoxel(std::get<1>(it), targetLeaf, offsets);
+
+#if OPENVDB_ABI_VERSION_NUMBER >= 6
+            sourceArray.loadData();
+            SingleCopyIterator singleCopyIterator;
```

Comment:
could move the construction of SingleCopyIterator out of this loop

Reply:
I'd be very surprised if the compiler doesn't do this optimisation. Reason for putting it in the loop is to reduce the number of #if statements littering the code without incurring unused variable warnings.

Reply:
Sure, this is minor, I think it's clearer regardless of what we assume the compilers doing (at the cost of more #if statements) but I'll leave it up to you.